djangocms_versioning/admin.py

```diff
@@ -433,38 +432,29 @@ def edit_redirect_view(self, request, object_id):
             raise Http404
 
         # Redirect
-        # If the object is editable the cms editable view should be used, with the toolbar.
-        if is_editable_model(version.content.__class__):
-            url = get_object_edit_url(version.content)
-        # Or else, the standard edit view should be used
-        else:
-            url = reverse('admin:{app}_{model}_change'.format(
-                app=version.content._meta.app_label,
-                model=version.content._meta.model_name,
-            ), args=(version.content.pk,))
-        return redirect(url)
+        return redirect(get_editable_url(version))
 
-    def revert_redirect_view(self, request, object_id):
+    def revert_view(self, request, object_id):
         """Redirects to the admin change view and creates a draft version
         if no draft exists yet.
         """
         version = self.get_object(request, unquote(object_id))
-        version = version.copy(request.user)
 
         if version is None:
             raise Http404
 
+        pks_for_grouper = version.versionable.for_grouper(
+            version.grouper).values_list('pk', flat=True)
+
+        drafts = Version.objects.filter(
+            object_id__in=pks_for_grouper, content_type=version.content_type,
+            state=DRAFT)
+
+        if not drafts.exists() and version.state in (UNPUBLISHED, ARCHIVED):
```

Comment:
Add more error handling, currently you'll redirect to old version if there's an existing draft.